src/core/ext/gcp/metadata_query.h

```diff
@@ -0,0 +1,64 @@
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef GRPC_SRC_CORE_EXT_GCP_METADATA_QUERY_H
+#define GRPC_SRC_CORE_EXT_GCP_METADATA_QUERY_H
+
+#include <grpc/support/port_platform.h>
+
+#include <string>
+
+#include "absl/functional/any_invocable.h"
+#include "absl/status/status.h"
+
+#include "src/core/lib/http/parser.h"
+#include "src/core/lib/iomgr/closure.h"
+#include "src/core/lib/iomgr/polling_entity.h"
+
+namespace grpc_core {
+
+// Fire and Forget class (Cleans up after itself.)
+// Fetches the value of an attribute from the MetadataServer on a GCP
+// environment.
+class MetadataQuery {
```

Comment:
The existing `MetadataQuery` object used by the google-c2p resolver is cancellable, and e.g. the resolver will cancel outstanding MD server queries during shutdown.

Looks like cancellation is removed from the revised API, is that intentional?

Reply:
MetadataQuery requests are intended to be quick enough that I don't think that the cancellation matters so much but it is also trivial enough to add, so I don't mind adding it.

Reply:
Done